src/libfetchers/path.cc

```diff
@@ -140,6 +141,20 @@ struct PathInputScheme : InputScheme
                 mtime = dumpPathAndGetMtime(absPath.string(), sink, defaultPathFilter);
             });
             storePath = store->addToStoreFromDump(*src, "source");
+        } else {
+            // To avoid copying the path again to the /nix/store, we need to add a cache entry.
+            ContentAddressMethod method = ContentAddressMethod::Raw::NixArchive;
+            auto fp = getFingerprint(store, input);
+            if (fp) {
+              auto cacheKey = fetchers::Cache::Key{"fetchToStore", {
+                  {"name", "source"},
```

Comment:
This should probably be
```suggestion
                  {"name", input.getName()},
```
but it looks like we're ignoring the `name` attribute elsewhere in the path fetcher.